src/editors/select.js

```diff
@@ -331,7 +331,8 @@ JSONEditor.defaults.editors.select = JSONEditor.AbstractEditor.extend({
       }
 
       // If the previous value is still in the new select options, stick with it
-      if(select_options.indexOf(prev_value) !== -1) {
+      // We have not found a case where else is preferable since it breakes when you reorder the enum source
+      if(true) {
```

Comment:
As a clarification on my PR descrition: If a clever condition can be used instead of `true` here, the PR could probably be merged. Maybe an explicit option would do.